### PR TITLE
update one product build package logic

### PR DESCRIPTION
**Pull request by cynthiajoan**
[replace this line]: # (Describe your changes in detail.)
***

[replace this line]: # (Describe your testing in detail.)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: unity_packer/debug_single_export_json/analytics.json [hunk 1]
```diff
@@ -0,0 +1,397 @@
+{
```
Comment (a-maurice) [inline unity_packer/debug_single_export_json/analytics.json:1]: Are all of these just the usual export, but with just App and the product they care about?  It feels bad having to maintain an extra set of files for this.

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -49,6 +50,22 @@ jobs:
     strategy:
       fail-fast: false
     steps:
+      - name: Check input
```
[the remark this answers is not in the document]
Comment (cynthiajoan) [reply .github/workflows/package.yml:53]: done

File: scripts/build_scripts/build_package.py [hunk 3]
```diff
@@ -141,6 +178,22 @@ def main(argv):
   if os.path.exists(output_folder):
     shutil.rmtree(output_folder)
 
+  if FLAGS.apis:
+    # This should only happen for debug purpose, to create packages for certain products
+    # Codes below should not run
```
[the remark this answers is not in the document]
Comment (cynthiajoan) [reply scripts/build_scripts/build_package.py:183]: if the apis are passed in, it should skip the rest of the packaging steps going below. I can also do an else if that feels more exclusive.

File: unity_packer/debug_single_export_json/analytics.json (hunk 1 above)
Comment (cynthiajoan) [reply unity_packer/debug_single_export_json/analytics.json:1]: Yes the export jsons are like what you described. I used a script to generate those so should not be too much maintain work. I also upload that script if that's better.

Comment (a-maurice) [reply unity_packer/debug_single_export_json/analytics.json:1]: I think it is fine for now. In the future we can look into making it use the main one, but we do rarely touch these files in general.

File: scripts/build_scripts/build_package.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (a-maurice) [reply scripts/build_scripts/build_package.py:183]: Maybe just rephrase it to:
If told to only build a subset, package just those products and exit early.

File: .github/workflows/package.yml [hunk 4]
```diff
@@ -49,6 +50,24 @@ jobs:
...
+        # This check the apis input. It detect whether the package is only asking for subset of
+        # of the products, like auth,firestore only etc. If the input apis are less than the
+        # default_apis env attribute, we determine this package only package subset and for 
+        # check up purpose only.
```
Comment (a-maurice) [inline .github/workflows/package.yml:57]: Instead of "we determine..."  maybe "we package only the provided apis, for test purposes only."

File: scripts/build_scripts/build_package.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cynthiajoan) [reply scripts/build_scripts/build_package.py]: done

File: .github/workflows/package.yml (hunk 4 above)
Comment (cynthiajoan) [reply .github/workflows/package.yml]: done

**Review by a-maurice — APPROVED** [review]
